utils/utils.go

```diff
@@ -573,8 +574,7 @@ func ConvertComponentIdIntoFriendlyPathSearch(id string) (string, string) {
 
 	// check for strange spaces, chars and if found, wrap them up, clean them and create a new cleaned path.
 	for i := range segs {
-		pathCharExp, _ := regexp.MatchString("[%=;~.]", segs[i])
-		if pathCharExp {
+		if pathCharExp.Match([]byte(segs[i])) {
```

Comment:
Nice! yeah all these regexes should always be pre-compiled, I must have been rushing.